_docs/policies/tests.md

```diff
@@ -0,0 +1,47 @@
+# `status-go` Test Policy
+
+- [Creating Tests](#creating-tests)
+- [Flaky Tests](#flaky-tests)
+
+## Creating Tests
+
+- All new functionality MUST be introduced with tests that:
+  - Prove that the functionality performs as described
+  - Can be falsified
+  - Are resistant to fuzzing
+- All new tests MUST BE validated via 1000 local tests. Ensuring that the test runs and passes consistently every time gives confidence that the test is not flaky.
```

Comment:
Would be great to automate it and run on CI. What is not a CI check, will never be respected. Some will forget, some will not know (new contributors), some will not run to save time.

Reply:
I agree with your points. For now we can include it as part of the policy as this is the behaviour we want regardless of how we enforce the requirement.

I'll raise an issue to explore / implement the automation of "new integration test mega runs". I'll track it via the Go Council project board ... Which doesn't exist yet but we probably should have it as we have multiple improvements to track, and this improvement is dependent on being able to categorise certain tests as "integration test".

Reply:
@Samyoul make issue and somewhere to track Go Council related things

Reply:
Ok this issue is now done https://github.com/status-im/status-go/issues/4952